### PR TITLE
Add docs clarifying behavior of retry files

### LINKED ISSUE #64564: retry files management
##### SUMMARY
There is no clear description as to how retry files are managed once a playbook is run with `--limit @/retry/file`

Say I run a playbook with a lot of hosts that might be *unreachable*. First time it runs there will be a retry file created. Second run will be run against the remaining hosts in the retry file.

Q1: Will the successfull hosts at the second run be "removed" from the retry file and only have the rest of the *unreachables*? Or, to put it differently, will there be another retry file created containing only the *unreachable* hosts after the second run? Similarly for the third, fourth, n-th run.

Q2: What happens to the retry file if ALL the hosts within are successfull? Is it removed/updated/etc?

##### ISSUE TYPE
- Documentation Report

##### COMPONENT NAME
C.RETRY_FILES_ENABLED

##### ANSIBLE VERSION
```
ansible 2.8.5
  config file = /path/to/config/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  7 2019, 00:51:29) [GCC 4.8.5 20150623 (Red Hat 4.8.5-39)]
```

##### CONFIGURATION
```
ACTION_WARNINGS(/path/to/config/ansible.cfg) = False
ANSIBLE_NOCOWS(/path/to/config/ansible.cfg) = True
ANSIBLE_PIPELINING(/path/to/config/ansible.cfg) = True
CACHE_PLUGIN(/path/to/config/ansible.cfg) = jsonfile
CACHE_PLUGIN_CONNECTION(/path/to/config/ansible.cfg) = /etc/ansible/cache
CACHE_PLUGIN_TIMEOUT(/path/to/config/ansible.cfg) = 86400
COMMAND_WARNINGS(/path/to/config/ansible.cfg) = False
DEFAULT_CALLBACK_WHITELIST(/path/to/config/ansible.cfg) = [u'timer', u'log_plays']
DEFAULT_FORKS(/path/to/config/ansible.cfg) = 10
DEFAULT_GATHERING(/path/to/config/ansible.cfg) = smart
DEFAULT_HOST_LIST(/path/to/config/ansible.cfg) = [u'/path/to/config/hosts']
DEFAULT_LOG_PATH(/path/to/config/ansible.cfg) = /var/log/ansible.log
DEFAULT_ROLES_PATH(/path/to/config/ansible.cfg) = [u'/etc/ansible/roles', u'/usr/share/ansible/roles']
HOST_KEY_CHECKING(/path/to/config/ansible.cfg) = False
RETRY_FILES_ENABLED(/path/to/config/ansible.cfg) = True
```

##### OS / ENVIRONMENT
OS: CentOS Linux release 7.7.1908 (Core)

##### ADDITIONAL INFORMATION
I cannot seem to find this information anywhere! No official documentation versus different outcomes in practice.

### PR DESCRIPTION
**Pull request by samdoran**
##### SUMMARY

Fixes #64564

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

`docs/docsite/rst/user_guide/intro_patterns.rst`
`lib/ansible/config/base.yml`

### CODE REVIEW

File: docs/docsite/rst/user_guide/intro_patterns.rst [hunk 1]
```diff
@@ -153,6 +153,10 @@ Finally, you can use ``--limit`` to read the list of hosts from a file by prefix
 
     ansible-playbook site.yml --limit @retry_hosts.txt
 
+If ``RETRY_FILES_ENABLED`` is set to ``True``, a ``.retry`` file will be created after the ``ansible-playbook`` run containing a list of failed hosts from all plays. This file is overwritten each time ``ansible-playook`` is finishing running.
```
Comment (samdoran) [inline docs/docsite/rst/user_guide/intro_patterns.rst]: Is there a way to link to the [this HTML page](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#retry-files-enabled) from within the RST?

Comment (samccann) [reply docs/docsite/rst/user_guide/intro_patterns.rst]: Yes. If you look at the reference_appendices/config.rst file (if you've done a make webdocs locally it should be there), you'll see every option is preceded by an .rst anchor that matches the  option name. So in this instance, :ref:`RETRY_FILES_ENABLED` should work.

Comment (samccann) [inline docs/docsite/rst/user_guide/intro_patterns.rst]:
```suggestion
If ``RETRY_FILES_ENABLED`` is set to ``True``, a ``.retry`` file will be created after the ``ansible-playbook`` run containing a list of failed hosts from all plays. This file is overwritten each time ``ansible-playook``  finishes running.
```

Comment (samdoran) [reply docs/docsite/rst/user_guide/intro_patterns.rst]: That's perfect. Thanks!

**Review by samccann — APPROVED** [review]

**Pull request merged**